leaderelection/config.go

```diff
@@ -46,10 +46,15 @@ func NewConfigFromMap(data map[string]string) (*Config, error) {
 	config := defaultConfig()
 
 	if err := cm.Parse(data,
+		// Parse legacy keys first
 		cm.AsDuration("leaseDuration", &config.LeaseDuration),
 		cm.AsDuration("renewDeadline", &config.RenewDeadline),
 		cm.AsDuration("retryPeriod", &config.RetryPeriod),
 
+		cm.AsDuration("lease-duration", &config.LeaseDuration),
+		cm.AsDuration("renew-deadline", &config.RenewDeadline),
+		cm.AsDuration("retry-period", &config.RetryPeriod),
```

Comment:
Same comment as in Serving: Should we be "smart" in our `cm.AsX` functions and auto-transform the keys between camelCase and kebab-case to allow both styles?

Reply:
I'm mixed on it - I figure a post-install job would help with the migration